modules/RTC/JitsiLocalTrack.js

```diff
@@ -320,6 +335,99 @@ export default class JitsiLocalTrack extends JitsiTrack {
         }
     }
 
+    /**
+     * Starts or stops the effect process and returns the corresponding stream.
+     *
+     * @private
+     * @param {Boolean} enable - Used to start or stop the effect processing
+     * @param {*} effect - Optional parameter represents effect class to create instance
+     * @returns {MediaStream}
+     */
+    _streamEffectEnable(enable: Boolean, effect) {
+        if (enable) {
+            this._effect = effect;
+            this._originalStream = this.stream;
+            this._streamEffect = new this._effect(this._originalStream);
+            this._streamEffect.startEffect();
+
+            return this._streamEffect.getStreamWithEffect();
+        }
+
+        this._streamEffect.stopEffect();
+        this._streamEffect = null;
+
+        return this._originalStream;
+    }
+
+    /**
+     * Switches the video stream associated with this track and updates the containers.
+     *
+     * @private
+     * @param {MediaStream} newStream - The MediaStream to replace
+     * @returns {Promise}
+     */
+    _switchStream(newStream: MediaStream) {
+        if (this._streamChangeInProgress === true) {
+            return Promise.reject(new Error('Stream change already in progress!'));
+        }
+
+        this._streamChangeInProgress = true;
+
+        if (this.isMuted()) {
+            return Promise.resolve();
+        }
+
+        let removeStreamPromise = Promise.resolve();
+
+        if (this.conference) {
+            removeStreamPromise = this.conference._removeLocalTrackAsMute(this);
+        }
+
+        return removeStreamPromise.then(() => {
+            this._setStream(newStream);
+
+            if (!this.isAudioTrack()) {
+                this.containers.forEach(
+                    cont => RTCUtils.attachMediaStream(cont, this.stream));
+            }
+
+            return this._addStreamToConferenceAsUnmute();
```

Comment:
`_removeLocalTrackAsMute` and `_addStreamToConferenceAsUnmute` are specifically designed for the video mute use case. Maybe we can create a new method that will replace the webrtc stream for a JitsiLocalTrack. WDYT? Probably we can do this even without Offer/Answer cycle.

Reply:
I could also use removeTrack and addTrack from JitsiConference to get the same result

Reply:
removeTrack and addTrack is fine for now. As further optimisation we can create JingleSession method that replaces a stream without doing Offer Answer. But let's address this trough another PR when we have more time.